code-challenges.test.js

```diff
@@ -15,42 +15,146 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fibLength1 = 6
+//const fibLength1 = 6
 // Expected output: [1, 1, 2, 3, 5, 8]
 
-const fibLength2 = 10
+//const fibLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+describe ("Fibonacci sequence", () => {
+    const fibLength1 = 6
+    const fibLength2 = 10
+    it ("should return an array of the Fibonacci sequence", () => {
+    expect(fibonacci(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+    expect(fibonacci(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+}) 
+
+// ReferenceError: fibonacci is not defined
 
 // b) Create the function that makes the test pass.
 
+    const fibonacci = (num) => {
+        let fibArray = [1, 1]
+        for (let i = 2; i < num; i++) {
+            fibArray.push(fibArray[i - 2] + fibArray[i - 1])
+        }
+        return fibArray
+    }
+
+//psuedo code:
+// first create a function
+// create a variable called fibArray and set it to an array with the first two numbers in the sequence
+// create a for loop that will run until the length of the array is equal to the number passed in
+// create a variable called i and set it to 2
+// create a variable called fibNum and set it to the sum of the previous two numbers in the array
+// push the fibNum into the array
+// return the array
+
+
+
+
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+//const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 
-const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+//const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+describe ("Odd numbers", () => {
+    const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+    it ("should return an array of only odd numbers sorted from least to greatest", () => {
+    expect(oddNums(fullArr1)).toEqual([-9, 7, 9, 199])
+    expect(oddNums(fullArr2)).toEqual([-823, 7, 23])
+    })
+}
+)
+
+//ReferenceError: oddNums is not defined
+
+
 
 // b) Create the function that makes the test pass.
 
+    const oddNums = (arr) => {
+        let oddArray = []
+        for (let i = 0; i < arr.length; i++) {
+            if (typeof arr[i] === "number" && arr[i] % 2 !== 0) {
+                oddArray.push(arr[i])
+            }
+        }
+        return oddArray.sort((a, b) => a - b)
+    }
```

Comment:
Logic, test and pseudocode all look excellent!  Try using higher order functions for problems like this if you can in the future.  Filter can do a lot of the hard work for you with this kind of problem.